pymc_experimental/statespace/core/statespace.py

```diff
@@ -0,0 +1,441 @@
+            pm.Deterministic("filtered_states", filtered_states)
+            pm.Deterministic("predicted_states", predicted_states)
+
+            pm.Deterministic("predicted_covariances", predicted_covariances)
+            pm.Deterministic("filtered_covariances", filtered_covariances)
+
+            pm.Potential("log_likelihood", log_likelihood)
+
+    def build_smoother_graph(self, mode=None):
+        pymc_model = modelcontext(None)
+        with pymc_model:
+            *_, T, Z, R, H, Q = self.unpack_statespace()
+            det_names = [x.name for x in pymc_model.deterministics]
+            if "filtered_states" not in det_names or "filtered_covariances" not in det_names:
+                raise ValueError(
+                    "Couldn't find Kalman filtered time series among model deterministics. Have you run"
+                    ".build_statespace_graph() ?"
+                )
+            fs_idx = det_names.index("filtered_states")
+            fc_idx = det_names.index("filtered_covariances")
+
+            filtered_states = pymc_model.deterministics[fs_idx]
+            filtered_covariances = pymc_model.deterministics[fc_idx]
+
+            smooth_states, smooth_covariances = self.kalman_smoother.build_graph(
+                T, R, Q, filtered_states, filtered_covariances, mode=mode
+            )
+
+            pm.Deterministic("smoothed_states", smooth_states)
+            pm.Deterministic("smoothed_covariances", smooth_covariances)
+
+    @staticmethod
+    def sample_conditional_prior(
+        filter_output="filtered", n_simulations=100, prior_samples=500
+    ) -> ArrayLike:
+        """
+        Sample from the conditional prior; that is, given parameter draws from the prior distribution, compute kalman
+        filtered trajectories. Trajectories are drawn from a single multivariate normal with mean and covariance
+        computed via either the kalman filter, smoother, or predictions.
+
+        Parameters
+        ----------
+        filter_output: string, default = 'filtered'
+            One of 'filtered', 'smoothed', or 'predicted'. Corresponds to which Kalman filter output you would like to
+            sample from.
+        n_simulations: int, default = 100
+            The number of simulations to run for each prior parameter sample drawn. Total trajectories returned by this
+            function will be n_simulations x prior_samples
+        prior_samples: int, default = 500
+            The number of samples to draw from the prior distribution, passed to pm.sample_prior_predictive. Defaults
+            to the PyMC default of 500.
+
+        Returns
+        -------
+        simulations: ArrayLike
+            A numpy array of shape (n_simulations x prior_samples, n_timesteps, n_states) with simulated trajectories.
+
+        """
+        validate_filter_arg(filter_output)
+        pymc_model = modelcontext(None)
+        with pymc_model, catch_warnings():
+            simplefilter("ignore", category=UserWarning)
```

Comment:
Why this blank `ignore` in a couple of places?

Reply:
It silences the user warning you get for sampling with a potential. I'm only sampling from the parameter priors then computing the posterior predictive myself, so it's not relevant.  It can be removed after the refactor to a random variable.